app/controllers/courses_controller.rb

```diff
@@ -41,7 +41,16 @@ def update
     validate
     handle_course_announcement(@course.instructors.first)
     slug_from_params if should_set_slug?
-    @course.update course_params
+
+    if params[:course].key?(:home_wiki)
```

Comment:
This new data handling should be done in a helper method, like with the timeline and slug routines.